src/core/Service.ts

```diff
@@ -91,20 +101,17 @@ export default class Service extends EventEmitter {
 
     this.ros.callOnConnection(call);
   }
-  /**
-   * @callback advertiseCallback
-   * @param {TRequest} request - The service request.
-   * @param {Partial<TResponse>} response - An empty dictionary. Take care not to overwrite this. Instead, only modify the values within.
-   * @returns {boolean} true if the service has finished successfully, i.e., without any fatal errors.
-   */
   /**
    * Advertise the service. This turns the Service object from a client
    * into a server. The callback will be called with every request
    * that's made on this service.
    *
-   * @param {advertiseCallback} callback - This works similarly to the callback for a C++ service and should take the following params
+   * @param callback This works similarly to the callback for a C++ service in that you should take care not to overwrite the response object.
+   *  Instead, only modify the values within.
    */
-  advertise(callback) {
+  advertise(
+    callback: (request: TRequest, response: Partial<TResponse>) => boolean,
+  ) {
```

Comment:
Specify type for API clarity
```suggestion
  ): Promise<void> {
```

Reply:
The same for other public methods